ledger/src/blockstore_processor.rs

```diff
@@ -1574,6 +1582,26 @@ pub fn fill_blockstore_slot_with_ticks(
     last_entry_hash
 }
 
+/// Check the transaction execution results to see if any instruction errored by exceeding the
+/// accounts data budget.  If yes, the whole block needs to be failed.
+fn check_accounts_data_budget<'a>(
+    execution_results: impl IntoIterator<Item = &'a TransactionExecutionResult>,
+) -> Result<()> {
+    if execution_results.into_iter().any(|execution_result| {
+        matches!(
+            execution_result.flattened_result(),
+            Err(TransactionError::InstructionError(
+                _,
+                InstructionError::AccountsDataBudgetExceeded
+            )),
+        )
+    }) {
+        Err(TransactionError::WouldExceedMaxTotalAccountDataLimit)
+    } else {
```

Comment:
Note that generating `InstructionError::AccountsDataBudgetExceeded` errors is behind a feature gate, so this `if` will always return false.

I've put a check around calling this function to see if the feature is enabled, so as to not take the hit searching the results until the feature is enabled.

However, within the context of `execute_batch()`, can/should I make this call to `.any()` with a parallel iterator?

Reply:
I don't see the need for a parallel iterator here, these batches are generally small